### PR TITLE
distrobox: 1.7.2.1 -> 1.8.0

### PR DESCRIPTION
**Pull request by pongo1231**
https://github.com/89luca89/distrobox/releases/tag/1.8.0

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/virtualization/distrobox/default.nix [hunk 1]
```diff
@@ -21,7 +21,7 @@ stdenvNoCC.mkDerivation (finalAttrs: {
 
     # https://github.com/89luca89/distrobox/issues/408
     substituteInPlace ./distrobox-generate-entry \
-      --replace-fail 'icon_default="''${HOME}/.local' "icon_default=\"$out"
+      --replace-fail 'icon_default="''${XDG_DATA_HOME:-''${HOME}/.local' 'icon_default="''${XDG_DATA_HOME:-''$out'
```
Comment (Scrumplex) [inline pkgs/applications/virtualization/distrobox/default.nix:24]: The literal result of this is:

```bash
icon_default="${XDG_DATA_HOME:-$out/share}/icons/terminal-distrobox-icon.svg"
```

In general, I don't really think we want to inject a Nix store path here. Why don't we just patch this to use the icon name (equivalent to having `Icon=terminal-distrobox-icon` in the final desktop entry)

Comment (Scrumplex) [reply pkgs/applications/virtualization/distrobox/default.nix:24]: Nevermind, I see that the remaining code assumes that this variable is a path.

I think this whole file should just be patched to resolve the fallback icon using its name.

Comment (Scrumplex) [reply pkgs/applications/virtualization/distrobox/default.nix:24]: I think this should work:

```patch
diff --git a/pkgs/applications/virtualization/distrobox/default.nix b/pkgs/applications/virtualization/distrobox/default.nix
index f79d09dc94b3..ce3c4e075a95 100644
--- a/pkgs/applications/virtualization/distrobox/default.nix
+++ b/pkgs/applications/virtualization/distrobox/default.nix
@@ -16,12 +16,14 @@ stdenvNoCC.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [ makeWrapper ];
 
+  patches = [
+    # https://github.com/89luca89/distrobox/issues/408
+    ./relative-default-icon.patch
+  ];
+
   installPhase = ''
     runHook preInstall
 
-    # https://github.com/89luca89/distrobox/issues/408
-    substituteInPlace ./distrobox-generate-entry \
-      --replace-fail 'icon_default="''${XDG_DATA_HOME:-''${HOME}/.local' 'icon_default="''${XDG_DATA_HOME:-''$out'
     ./install -P $out
 
     runHook postInstall
diff --git a/pkgs/applications/virtualization/distrobox/relative-default-icon.patch b/pkgs/applications/virtualization/distrobox/relative-default-icon.patch
new file mode 100644
index 000000000000..069c2d40d618
--- /dev/null
+++ b/pkgs/applications/virtualization/distrobox/relative-default-icon.patch
@@ -0,0 +1,26 @@
+diff --git a/distrobox-generate-entry b/distrobox-generate-entry
+index 65fcea0..8d57e4e 100755
+--- a/distrobox-generate-entry
++++ b/distrobox-generate-entry
+@@ -51,7 +51,7 @@ container_manager="autodetect"
+ container_name_default="my-distrobox"
+ delete=0
+ icon="auto"
+-icon_default="${XDG_DATA_HOME:-${HOME}/.local/share}/icons/terminal-distrobox-icon.svg"
++icon_default="terminal-distrobox-icon"
+ verbose=0
+ online=0
+ version="1.8.0"
+@@ -335,12 +335,6 @@ if [ "${icon}" = "auto" ]; then
+ 
+ 	icon_url="$(echo "${DISTRO_ICON_MAP}" | grep "${container_distro}:" | cut -d':' -f2-)"
+ 
+-	# Distro not found in our map, fallback to generic icon
+-	if [ -z "${icon_url}" ]; then
+-		icon_url="https://raw.githubusercontent.com/89luca89/distrobox/main/icons/terminal-distrobox-icon.svg"
+-		container_distro="terminal-distrobox-icon"
+-	fi
+-
+ 	if [ -n "${icon_url}" ] && [ "${download}" != "null" ]; then
+ 		icon_extension="${icon_url##*.}"
+ 
```

**Comment by pongo1231** [comment]
Added the patch, thanks

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**